Flaky test common::score_fusion::tests::welford_calc_vs_naive

Failed on Windows CI https://github.com/qdrant/qdrant/actions/runs/16044854927/job/45273637563?pr=6799

```
 TRY 2 FAIL [   0.383s] segment common::score_fusion::tests::welford_calc_vs_naive
  stdout ───

    running 1 test
    test common::score_fusion::tests::welford_calc_vs_naive ... FAILED

    failures:

    failures:
        common::score_fusion::tests::welford_calc_vs_naive

    test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 570 filtered out; finished in 0.36s
    
  stderr ───

    thread 'common::score_fusion::tests::welford_calc_vs_naive' panicked at lib\segment\src\common\score_fusion.rs:185:9:
    3393.2866 is not close to 3393.2974
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

    thread 'common::score_fusion::tests::welford_calc_vs_naive' panicked at lib\segment\src\common\score_fusion.rs:185:9:
    3393.2866 is not close to 3393.2974

    thread 'common::score_fusion::tests::welford_calc_vs_naive' panicked at lib\segment\src\common\score_fusion.rs:185:9:
    3393.2866 is not close to 3393.2974

    thread 'common::score_fusion::tests::welford_calc_vs_naive' panicked at lib\segment\src\common\score_fusion.rs:185:9:
    3393.2866 is not close to 3393.2974

...

    thread 'common::score_fusion::tests::welford_calc_vs_naive' panicked at lib\segment\src\common\score_fusion.rs:185:9:
    3393.2866 is not close to 3393.2974
    proptest: Aborting shrinking after the PROPTEST_MAX_SHRINK_ITERS environment variable or ProptestConfig.max_shrink_iters iterations (set 1024 to a large(r) value to shrink more; current configuration: 1024 iterations)

    thread 'common::score_fusion::tests::welford_calc_vs_naive' panicked at lib\segment\src\common\score_fusion.rs:188:5:
    Test failed: 3393.2866 is not close to 3393.2974.
    minimal failing input: scores = [
        -89.694244,
        68.54236,
        -46.20981,
        39.62516,
        -12.572493,
        3.2352805,
        -80.22215,
        87.94728,
        52.740093,
        85.23319,
        -65.185165,
        -73.220764,
        -92.93057,
        -54.780697,
        92.01279,
        52.644333,
        87.1816,
        -83.749245,
        58.918304,
        -38.682686,
        -15.217656,
        56.48435,
        -58.317757,
        59.8789,
        6.191427,
        30.41261,
        64.56317,
        -20.80256,
        21.078533,
        -50.606934,
        31.77195,
        -82.71065,
        -36.005642,
        2.1219108,
        2.8251493,
        67.91865,
        -88.15664,
        -62.56238,
        -77.741936,
        72.71868,
        10.056342,
        -94.645134,
        39.93914,
        -80.66441,
        21.712961,
        98.631226,
        70.22569,
        26.809128,
        -98.973366,
        62.300976,
        -87.14279,
        82.81035,
        19.56247,
        -31.72698,
        28.97621,
        -75.40742,
        -60.763973,
        45.100235,
        -39.545868,
        9.835315,
        -29.452227,
        -24.542707,
        -34.455917,
        42.355984,
        55.31158,
        93.046,
        -40.862503,
        36.663544,
        88.13288,
        83.46571,
        -35.913,
        50.056202,
        -4.879256,
        -94.223526,
        -96.18551,
        77.468575,
        -91.18846,
        -95.61745,
        -77.523155,
        82.7762,
        -5.092673,
        88.84102,
        61.420204,
        -15.461961,
        -59.771843,
        57.408924,
        5.7903666,
        84.80806,
        -62.75431,
        -90.57634,
        -21.34188,
        42.614124,
        15.580222,
        -35.206818,
        -29.943792,
        -43.730537,
        -71.01367,
        -94.366585,
        72.70432,
        8.3603945,
        0.65545064,
        1.6082511,
        -6.093136,
        -66.53697,
        -17.735987,
        60.337696,
        -91.64411,
        -91.668365,
        -60.102528,
        23.287083,
        -30.38929,
        -43.04157,
        -23.70972,
        79.382866,
        -76.24054,
        1.5598285,
        65.59531,
        97.622604,
        -16.066595,
        -61.13603,
        -57.725307,
        -1.9734721,
        32.50394,
        -9.399988,
        0.47435454,
        -41.115738,
        -75.30547,
        46.012558,
        31.666658,
        34.721195,
        11.549813,
        -70.77696,
        50.559082,
        90.07843,
        10.899997,
        -1.9432266,
        -38.596973,
        -41.908165,
        14.737823,
        -44.034565,
        -67.685974,
        22.074898,
        65.11266,
        -27.096622,
        56.13967,
        -98.36629,
        35.054825,
        -3.7687447,
        -30.214453,
        92.41021,
        81.281715,
        -26.375008,
        -3.7421205,
        -65.386116,
        -26.39815,
        -13.756069,
        -23.75232,
        37.134724,
        39.454838,
        77.642715,
        -57.932907,
        -55.180237,
        -83.36304,
        -30.767492,
        44.304916,
        -37.409958,
        1.8033358,
        74.992615,
        8.386365,
        53.94418,
        85.6088,
        -60.581116,
        22.15182,
        -92.50607,
        54.305504,
        91.877754,
        -44.56966,
        64.6624,
        89.045204,
        30.462152,
        32.448425,
        -74.83976,
        -22.4349,
        90.708305,
        4.328706,
        88.88225,
        -94.51213,
        4.977593,
        -11.455499,
        -27.525684,
        -93.25648,
        -55.5969,
        -72.53144,
        -45.554043,
        15.457888,
        -15.621477,
        -1.4709384,
        -57.779137,
        16.47733,
        35.080196,
        -1.4872943,
        98.029816,
        -10.170027,
        60.10577,
        -39.238876,
        -47.83641,
        24.96546,
        79.18106,
        -72.24373,
        -62.125965,
        -41.114693,
        65.52517,
        46.905537,
        -2.821711,
        -87.425186,
        13.860518,
        77.684784,
        20.357565,
        99.57925,
        -93.51393,
        57.04015,
        34.92252,
        -32.264362,
        76.11047,
        -75.717285,
        12.452317,
        72.9141,
        -49.477146,
        -28.508759,
        24.419693,
        49.344227,
        25.986488,
        -27.572256,
        -22.432713,
        89.614426,
        75.02871,
        -3.8915713,
        -66.24823,
        81.047035,
        -69.99688,
        94.80065,
        -34.390717,
        -77.91061,
        94.44753,
        93.0339,
        -92.958,
        -54.296043,
        53.480774,
        49.947815,
        -7.924837,
        98.63043,
        -94.57219,
        -86.67263,
        85.21839,
        65.211334,
        -97.03482,
        -40.86591,
        -21.054476,
        -12.879465,
        -36.77088,
        85.11147,
        -28.568256,
        -95.350975,
        -88.11192,
        57.458588,
        91.23729,
        -96.71319,
        -77.70026,
        -69.54237,
        -20.32606,
        19.466703,
        15.316607,
        6.216384,
        -18.93144,
        94.43418,
        -88.625755,
        -47.339027,
        -42.183727,
        -27.716455,
        -97.28496,
        63.939056,
        53.975082,
        -29.248297,
        35.409233,
        43.419773,
        45.179173,
        33.28603,
        8.918921,
        -16.469275,
        33.37465,
        40.43299,
        88.71109,
        40.8253,
        46.90676,
        99.56681,
        -33.38388,
        -57.498726,
        -6.527813,
        -71.7851,
        -64.19899,
        28.701902,
        25.52938,
        74.47072,
        -51.271534,
        59.273186,
        -40.908108,
        24.453257,
        14.560201,
        62.522964,
        26.076862,
        7.816189,
        14.861689,
        -84.53185,
        -82.5781,
        -64.075485,
        -2.2465546,
        -95.47076,
        -37.754047,
        -89.67898,
        -78.14617,
        40.45494,
        -22.447039,
        -44.809658,
        34.945904,
        -25.275846,
        -48.0215,
        75.28735,
        -12.688054,
        1.0151354,
        -79.32819,
        -69.03814,
        -76.22499,
        79.814476,
        -58.198174,
        93.85107,
        -27.171991,
        -99.871,
        -39.437374,
        -90.21962,
        -43.15089,
        -69.061295,
        -95.85782,
        3.5515022,
        61.246628,
        12.649238,
        -24.709948,
        -30.705833,
        47.805496,
        -25.82868,
        -82.46912,
        -36.83097,
        96.48809,
        -82.12259,
        -28.796244,
        -91.45769,
        35.20454,
        7.172272,
        -73.618484,
        -72.1833,
        83.15013,
        -30.071285,
        -74.9563,
        -35.92604,
        59.11517,
        93.21339,
        51.810677,
        -82.640465,
        -54.36047,
        -27.625406,
        -84.14807,
        -74.42249,
        37.446945,
        -96.65392,
        -67.685974,
        32.564568,
        -24.853704,
        -28.185638,
        -75.77928,
        -28.590595,
        47.538506,
        -49.55052,
        -32.14857,
        -17.80959,
        -54.49459,
        88.01393,
        92.653275,
        19.014927,
        61.3015,
        -72.83358,
        48.636562,
        42.88966,
        -60.560696,
        92.59201,
        -15.383772,
        -31.15402,
        -20.573465,
        -49.992783,
        69.43106,
        26.552425,
        55.984795,
        -80.74724,
        -78.09635,
        -22.991634,
        -57.260735,
        -64.300896,
        -34.04837,
        -4.9391913,
        49.525455,
        -59.086994,
        72.62734,
        19.135538,
        99.206795,
        2.0772002,
        -52.28243,
        54.733376,
        -81.93355,
        -12.9132,
        -37.59752,
        -50.610706,
        -50.14638,
        -73.08764,
        25.153633,
        -13.641157,
        -36.444374,
        89.63563,
        66.52757,
        28.293789,
        -3.7820413,
        3.463018,
        0.21259072,
        -69.454414,
        -88.63836,
        -22.279005,
        43.69511,
        44.894016,
        -14.039931,
        -2.4093003,
        -20.079504,
        88.92665,
        -24.415607,
        98.82639,
        98.253555,
        -37.5515,
        -13.695729,
        -40.605316,
        -8.508059,
        80.07005,
        66.0477,
        -40.008724,
        38.654785,
        60.4363,
        95.85225,
        10.691567,
        -3.8940237,
        11.230054,
        83.455666,
        27.35422,
        78.52534,
        94.744644,
        -14.445926,
        -19.60719,
        -99.91217,
        17.140837,
        -91.764534,
        -9.422861,
        -45.304993,
        9.435237,
        -81.03823,
        18.68796,
        54.4031,
        -92.37142,
        -77.84898,
        -18.506865,
        98.88581,
        -22.474,
        87.26943,
        -76.56829,
        73.04583,
        69.20659,
        -19.440376,
        22.39687,
        -88.96234,
        -72.6736,
        -42.169838,
        38.16462,
        -52.19919,
        40.29815,
        -74.56591,
        74.73352,
        -36.731853,
        15.793263,
        -18.134848,
        -93.25935,
        88.6187,
        82.24063,
        57.656334,
        13.329074,
        -72.59516,
        -78.97023,
        94.77465,
        -67.70004,
        -99.054405,
        -55.85779,
        66.54195,
        -36.673706,
        -64.884476,
        40.092087,
        -47.21385,
        56.41079,
        -92.75354,
        -69.722404,
        -29.038002,
        -55.35015,
        -61.571407,
        61.65034,
        -65.243835,
        -97.30734,
        -30.905159,
        -97.04631,
        65.7533,
        -18.961058,
        -17.21077,
        12.892196,
        40.92707,
        20.23454,
        48.835754,
        27.13725,
        -13.71441,
        93.33344,
        -72.18071,
        96.2238,
        52.509674,
        9.98447,
        -92.75065,
        -78.73425,
        -77.94738,
        -67.36497,
        47.33109,
        -70.191536,
        59.073364,
        96.7841,
        24.828842,
        84.68185,
        78.8613,
        62.707928,
        84.63376,
        79.36071,
        -93.82562,
        47.201916,
        -50.950188,
        76.93604,
        -70.28588,
        97.84497,
        44.5005,
        -86.71572,
        46.19516,
        -39.321587,
        54.22667,
        34.49552,
        73.9727,
        -46.65523,
        39.31877,
        -99.79424,
        -60.846386,
        -17.026718,
        36.785183,
        68.474556,
        44.78309,
        -69.475426,
        -99.504456,
        -96.03172,
        93.39262,
        -8.397469,
        -98.515495,
        30.669184,
        -42.048443,
        3.8114538,
        -87.14907,
        69.9209,
        4.7167954,
        93.34212,
        -27.296087,
        22.444628,
        -42.01307,
        58.89129,
        -53.76983,
        55.66157,
        -7.2394166,
        -3.4230413,
        -32.724827,
        -25.978577,
        29.934925,
        33.24254,
        -61.906815,
        -22.358725,
        21.431684,
        -27.816004,
        -35.569397,
        -36.38072,
        62.50525,
        -90.863434,
        24.344122,
        91.4874,
        -37.690155,
        -53.285023,
        95.08102,
        92.74139,
        57.478065,
        -69.19287,
        -15.495776,
        56.255833,
        4.585658,
        -4.629286,
        77.239365,
        22.175749,
        -13.546028,
        24.961834,
        88.74195,
        -31.927385,
        4.950319,
        -13.307036,
        -5.5792747,
        68.82676,
        -0.15803248,
        53.31544,
        71.28564,
        22.141125,
        -57.737976,
        3.9740014,
        -40.731365,
        18.211905,
        98.34909,
        0.0014348504,
        44.375305,
        -29.748796,
        88.40794,
        -33.156975,
        80.59477,
        25.958916,
        -46.387363,
        72.35782,
        -87.913155,
        4.9267807,
        -80.621086,
        90.80141,
        -73.00285,
        53.17388,
        -98.19866,
        16.772367,
        -81.81509,
        -74.1436,
        -41.764103,
        -90.46795,
        -23.99452,
        79.32601,
        66.51257,
        2.8561482,
        44.934914,
        -45.631992,
        -89.73425,
        -20.676603,
        -43.17499,
        -51.61465,
        62.475307,
        -54.06499,
        -44.66275,
        -64.00062,
        21.741203,
        -17.775263,
        15.428359,
        7.478505,
        -48.121887,
        -47.433273,
        50.759693,
        86.12309,
        -62.414352,
        -97.45403,
        74.25907,
        -74.26233,
        -58.16471,
        54.700436,
        -71.02029,
        34.52298,
        88.61228,
        13.105981,
        85.22407,
        -28.915012,
        -78.43395,
        -80.07443,
        19.867825,
        -36.87074,
        -53.762367,
        4.2378826,
        17.985363,
        82.87688,
        36.987984,
        17.944607,
        -64.137436,
        -89.07444,
        -12.826922,
        -39.401733,
        28.59207,
        70.280716,
        -45.97226,
        2.4041216,
        75.08263,
        7.5444865,
        -16.594034,
        93.6603,
        -80.11761,
        -71.800446,
        -29.665447,
        10.848482,
        72.59764,
        -74.92173,
        -30.563026,
        -68.94109,
        -38.66073,
        38.77801,
        45.29459,
        -52.929256,
        -8.216714,
        -1.859577,
        27.467943,
        -82.944374,
        -67.36654,
        64.32916,
        -7.97085,
        62.222218,
        -60.61077,
        -51.377094,
        -88.63666,
        4.1514587,
        -6.383806,
        50.54857,
        64.926285,
        95.10497,
        -88.100075,
        43.71211,
        87.24592,
        20.206062,
        -25.350193,
        14.779552,
        -54.11726,
        -71.47663,
        -61.28376,
        -28.528921,
        2.3626993,
        22.753126,
        98.77136,
        83.77269,
        -67.56891,
        -36.776466,
        51.73047,
        -84.65287,
        -13.141012,
        -61.820587,
        -81.38004,
        45.936066,
        23.119425,
        12.152249,
        60.096806,
        -48.454742,
        -60.707886,
        -2.3468745,
        -73.20762,
        31.589561,
        59.50898,
        65.10969,
        -48.479176,
        38.382893,
        -95.49387,
        61.8681,
        -7.0237894,
        -77.87608,
        93.42077,
        30.463417,
        19.76628,
        -40.24059,
        0.8773079,
        26.507725,
        85.15634,
        -28.801485,
        63.62554,
        78.5431,
        -57.767143,
        -79.806244,
        -50.95956,
        -48.079617,
        -35.69518,
        68.575134,
        55.78903,
        25.39232,
        -72.56798,
        1.1250265,
        93.29041,
        18.140472,
        48.41555,
        -18.940277,
        72.61034,
        12.642013,
        -76.81076,
        81.63327,
        -86.16345,
        78.42405,
        76.61773,
        76.07745,
        0.49227384,
        76.97762,
        78.486595,
        -8.32983,
        -35.260426,
        86.91342,
        69.58965,
        -55.404827,
        -34.221664,
        16.055666,
        -41.543568,
        -29.146143,
        83.299614,
        91.619225,
        70.22614,
        -89.24363,
        14.47434,
        -51.908535,
        36.53611,
        68.61032,
        99.62494,
        73.96245,
        38.41338,
        -99.2053,
        52.2604,
        -74.91006,
        43.82279,
        17.06331,
        -98.76151,
        -13.040651,
        -89.3065,
        40.002,
        -86.27678,
        -43.28775,
        -93.1233,
        2.5932202,
        49.913082,
        -65.48551,
        -18.579058,
        -0.44772926,
        -94.00597,
        54.642025,
        8.091802,
        27.566769,
        -68.292656,
        90.12783,
        -28.630457,
        -75.8927,
        51.737213,
        -41.266815,
        48.537304,
        57.35733,
        58.919228,
        -19.218313,
        -3.3913515,
        -37.328323,
        33.21944,
        46.277454,
        -8.98436,
        0.639414,
        -76.734276,
        -5.606918,
        -29.72806,
        34.407146,
        -22.921612,
        15.788931,
        26.457142,
        -46.86084,
        -76.81096,
        67.679405,
        -75.51093,
        -19.216705,
        0.48545122,
        99.729935,
        -74.73819,
        64.37623,
        -85.94921,
        38.26844,
        -21.168365,
        -7.8421187,
        -40.40178,
        56.06638,
        9.513334,
        26.650366,
        -96.59973,
        57.812275,
        13.184344,
        63.259315,
        -4.7272053,
        20.69246,
        -60.644928,
        -13.620829,
        -40.98453,
        68.54232,
        -70.310196,
        40.769093,
        -40.191235,
        34.57515,
        94.71346,
        -73.84115,
        65.72624,
        -42.276405,
        34.838226,
        39.997555,
        -34.33709,
        15.868152,
        89.1077,
        -35.273037,
        0.13912147,
        -52.259647,
        94.11291,
        -63.905262,
        -17.158861,
        -28.581345,
        -58.95007,
        -37.835987,
        22.479052,
        94.12416,
        26.340311,
        -8.550474,
        -66.56001,
        19.958437,
        86.02857,
        -74.69597,
        -61.268482,
        79.27663,
        -46.466232,
        76.79968,
        -82.86537,
        78.82142,
        -2.1402924,
        32.78811,
        -5.2879505,
        -95.352036,
        -21.08824,
        -20.008766,
        -81.924706,
        -3.1899219,
        -45.926834,
        69.600845,
        48.830894,
        74.209854,
        92.92848,
        56.60148,
        34.773502,
        93.05091,
        10.037273,
        -24.615316,
        4.5575933,
        -5.9273205,
        -30.219824,
        -43.30441,
        53.173176,
        -60.837246,
        -24.28133,
        -60.504627,
        -43.479233,
        11.414343,
        -66.189995,
        -41.395634,
        -1.7193652,
        55.39853,
    ]
    	successes: 0
    	local rejects: 0
    	global rejects: 0
    

error: test run failed
Error: Process completed with exit code 1.
0s
```

--- a/lib/segment/src/common/score_fusion.rs
+++ b/lib/segment/src/common/score_fusion.rs
@@ -182,22 +182,17 @@ mod tests {
     }
 
     fn assert_close(a: f32, b: f32) {
-        // Round both values to 2 significant digits for comparison
-        let round_to_2_sig = |x: f32| -> f32 {
-            if x == 0.0 {
-                return 0.0;
-            }
-            let magnitude = x.abs().log10().floor();
-            let factor = 10.0_f32.powi(1 - magnitude as i32);
-            (x * factor).round() / factor
-        };
+        // Choose the more relaxed tolerance, absolute or relative based on the values.
+        let abs_tolerance = 1e-5f32;
+        let rel_tolerance = 1e-4f32;
 
-        let a_rounded = round_to_2_sig(a);
-        let b_rounded = round_to_2_sig(b);
+        let diff = (a - b).abs();
+        let max_val = a.abs().max(b.abs());
+        let tolerance = abs_tolerance.max(rel_tolerance * max_val);
 
-        assert_eq!(
-            a_rounded, b_rounded,
-            "{a} (rounded: {a_rounded}) is not close to {b} (rounded: {b_rounded}) when comparing 2 significant digits"
+        assert!(
+            diff <= tolerance,
+            "{a} is not close to {b}: difference {diff} exceeds tolerance {tolerance}"
         );
     }
 

